Source/BugsnagEvent.m

```diff
@@ -411,7 +415,7 @@ - (instancetype _Nonnull)initWithErrorName:(NSString *_Nonnull)name
         _errorMessage = message;
         _overrides = [NSDictionary new];
         self.metadata = [metadata deepCopy] ?: [BugsnagMetadata new];
-        _releaseStage = config.releaseStage;
+        self.releaseStage = config.releaseStage;
```

Comment:
Does it make sense to change other synthesized ivar accesses to dot notation now for consistency?

Reply:
I think it'd be best to avoid doing this in the constructor - I've only changed this for `releaseStage` as the side-effect of the setter is required. In this case the setter sets an 'override property' which is then persisted in the KSCrashReport and can be read when the report is delivered.

Previous to this changeset we set this via `event.releaseStage`, but now because we're enforcing `event.app.releaseStage`, we need to serialize the value in the same place in the KSCrash report. Using this felt like the simplest mechanism.

Reply:
Good explanation of the thinking behind the change, thanks.